reason-react-navigation-example/src/TwitterNavigation.re

```diff
@@ -0,0 +1,628 @@
+open ReactNavigation;
+open ReactNative;
+
+module Icons = {
+  let filter =
+    Image.Source.fromRequired(Packager.require("./Icons/filter.png"));
+
+  let settings =
+    Image.Source.fromRequired(Packager.require("./Icons/settings.png"));
+
+  let home = Image.Source.fromRequired(Packager.require("./Icons/home.png"));
+
+  let message =
+    Image.Source.fromRequired(Packager.require("./Icons/message.png"));
+
+  let notification =
+    Image.Source.fromRequired(Packager.require("./Icons/notification.png"));
+
+  let search =
+    Image.Source.fromRequired(Packager.require("./Icons/search.png"));
+
+  let compose =
+    Image.Source.fromRequired(Packager.require("./Icons/compose.png"));
+
+  let avatar =
+    Image.(
+      Source.fromUriSource(
+        uriSource(~uri="https://github.com/reasonml-community.png", ()),
+      )
+    );
+};
+
+type tweet = {
+  id: string,
+  text: string,
+  thread: bool,
+};
+
+let dummyTweets = [|
+  {id: "id_1", text: "Some insightful tweet", thread: false},
+  {id: "id_2", text: "Some insightful tweet", thread: true},
+  {id: "id_3", text: "Some insightful tweet", thread: false},
+  {id: "id_4", text: "Some insightful tweet", thread: true},
+  {id: "id_5", text: "Some insightful tweet", thread: false},
+  {id: "id_6", text: "Some insightful tweet", thread: true},
+  {id: "id_7", text: "Some insightful tweet", thread: false},
+  {id: "id_8", text: "Some insightful tweet", thread: true},
+  {id: "id_9", text: "Some insightful tweet", thread: false},
+  {id: "id_10", text: "Some insightful tweet", thread: true},
+|];
+
+let getTweet = (~id, onResult) =>
+  Js.Global.setTimeout(
+    () => onResult(dummyTweets->Belt.Array.keep(tweet => tweet.id == id)),
+    500,
+  );
+
+module ComposeRoute =
+  Route({
+    type params = unit;
+    let routeName = "Compose";
+  });
+
+module ComposeButton = {
+  [@react.component]
+  let make = (~onPress) =>
+    <TouchableOpacity
+      onPress
+      style=Style.(
+        style(
+          ~backgroundColor="#5D9FEF",
+          ~elevation=3.,
+          ~shadowRadius=3.,
+          ~shadowOpacity=0.3,
+          ~shadowOffset=offset(~height=2., ~width=0.),
+          ~position=`absolute,
+          ~right=16.->dp,
+          ~bottom=14.->dp,
+          ~alignItems=`center,
+          ~justifyContent=`center,
+          ~borderRadius=25.,
+          ~width=50.->dp,
+          ~height=50.->dp,
+          (),
+        )
+      )>
+      <Image
+        style=Style.(
+          style(~width=24.->dp, ~margin=11.->dp, ~height=24.->dp, ())
+        )
+        source=Icons.compose
+      />
+    </TouchableOpacity>;
+};
+
+let placeholder = (~text, ~navigation: navigation) =>
+  <>
+    <Centered> <Text> {text |> React.string} </Text> </Centered>
+    <ComposeButton
+      onPress={_ => navigation->Navigation.navigate("Compose")}
```

Comment:
This seems a bit unsafe isn't it? What happen if you made a typo in the string? red screen?